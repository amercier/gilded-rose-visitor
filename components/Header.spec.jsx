import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import ConnectedHeader, { Header } from './Header';

describe('Header', () => {
  it('renders the header', () => {
    const index = shallow(<Header cartLength={10} />);
    expect(index).toMatchSnapshot();
  });
});

describe('Header (Redux-connected)', () => {
  const mockStore = configureStore();

  it('uses values of cart object from the store', () => {
    const store = mockStore({
      cartReducer: { cart: ['a', 'b'] },
    });
    const index = mount(
      <Provider store={store}>
        <ConnectedHeader />
      </Provider>,
    );
    const indexComponent = index.find(Header).first();
    expect(indexComponent.props().cartLength).toEqual(2);
  });
});
