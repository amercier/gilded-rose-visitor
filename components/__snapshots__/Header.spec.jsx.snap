// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <Header__Container>
      <Link
        href="/"
      >
        <WithStyles(Button)
          color="inherit"
        >
          <WithStyles(Typography)
            color="inherit"
            noWrap={true}
            style={
              Object {
                "lineHeight": "64px",
              }
            }
            variant="h6"
          >
            Gilded Rose Inn
          </WithStyles(Typography)>
        </WithStyles(Button)>
      </Link>
      <Header__ButtonBar>
        <Link
          href="/cart"
        >
          <WithStyles(IconButton)
            color="inherit"
          >
            <WithStyles(Badge)
              badgeContent={10}
              color="secondary"
              invisible={false}
            >
              <pure(ShoppingCartIcon) />
            </WithStyles(Badge)>
          </WithStyles(IconButton)>
        </Link>
      </Header__ButtonBar>
    </Header__Container>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
